parsing/parser.mly

```
@@ -150,20 +151,21 @@ let neg_string f =
   then String.sub f 1 (String.length f - 1)
   else "-" ^ f
 
-let mkuminus ~oploc name arg =
+let mkuminus ~sloc ~oploc name arg =
   match name, arg.pexp_desc with
-  | "-", Pexp_constant(Pconst_integer (n,m)) ->
-      Pexp_constant(Pconst_integer(neg_string n,m))
-  | ("-" | "-."), Pexp_constant(Pconst_float (f, m)) ->
-      Pexp_constant(Pconst_float(neg_string f, m))
+  | "-", Pexp_constant({pconst_desc = Pconst_integer (n,m); _}) ->
+      Pexp_constant(mkconst ~loc:sloc (Pconst_integer(neg_string n, m)))
+  | ("-" | "-."), Pexp_constant({pconst_desc = Pconst_float (f, m); _}) ->
+      Pexp_constant(mkconst ~loc:sloc (Pconst_float(neg_string f, m)))
```

Here are with `mkuplus` below, I think that the record pattern on the `constant` record should list all fields. Otherwise if we add new fields later (say to support attributes) the new fields will be silently dropped.

In fact there is already a bug related to dropping information in these functions today, `-(1[@foo])` will drop the attribute.

---

Changed! I can have a look to the dropped attribute in an other PR.